### DISCUSSION TITLE
tailwindcss breakpoints not working with Nextjs 13 - app directory

### DISCUSSION THREAD

**Discussion by nguyenthinhkha** [tailwindlabs/tailwindcss#10413] [Help] [answered]
**What version of Tailwind CSS are you using?**

tailwindcss v3.2.4

**What build tool (or framework if it abstracts the build tool) are you using?**

Nextjs v13.1.5 - app directory

**What version of Node.js are you using?**

Node v18.12.1

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**Reproduction URL**

- New next js project with experimental app directory
- Set up tailwindcss
- In app/page.tsx
```
export default async function HomePage() {
  return (
    <div>
      <h1 className="text-3xl font-bold underline">Homepage</h1>
      <div className="md:hidden">Hello small factor</div>
    </div>
  );
}
```

**Describe your issue**

Text "Hello small factor" not show even screen size <768px

**Comment by nguyenthinhkha** [comment, marked as answer]
After searching through internet, I found that issue was solved by adding this line in head.tsx file
```
<meta name="viewport" content="width=device-width, initial-scale=1.0"></meta>
```
Hope this help!

**Reply by MAb540** [reply]
@nguyenthinhkha but i don't have any head.tsx, which file are you reffering?

**Reply by pedro-santos21** [reply]
@nguyenthinhkha No head.tsx here... You mean a HTML header?

**Comment by john-subba** [comment]
After adding meta tags you also want to check the content section of tailwind.config.js file and make sure it is pointing to right folder. I was so silly and was stuck in this silly issue for couple of hours.
New NextJS 13 Beta app supports meta tags like this: 
```
export const metadata = {
  viewport: {
    width: "device-width",
    height: "device-height",
    initialScale: 1,
    maximumScale: 1,
    minimumScale: 1,
  },
};

```

**Reply by florianzemma** [reply]
Thanks a lot,
My breakpoints were not triggered and I was going crazy but I have created a folder feature which is at the same level of the app folder so I had to add the path to this folder in tailwind config :)

**Reply by MAb540** [reply]
@john-subba in which file would i set this metadata config? do i need to add this in every page in next 13 app dir?

**Reply by john-subba** [reply]
@MAb540 you don't need to add this in every page. Adding only on layout.js file will be enough. But according to nextjs docs it is added by default. [Docs](https://nextjs.org/docs/app/building-your-application/optimizing/metadata)
